### PR TITLE
refactor: ignore partitions in table skipped compactions

### LINKED ISSUE #6657: Compactor2: Do not read (or recompact) partition that its in skipped_compactions
We currently do not check whether the partition is already in skipped_compactions and recomapct if we read it again. we should either do not read it or check before compact if it is in the skipped_compactions

### PR DESCRIPTION
**Pull request by NGA-TRAN**
Closes #6657

### CODE REVIEW

File: compactor2/src/driver.rs [hunk 1]
```diff
@@ -131,125 +143,3 @@ async fn try_compact_partition(
 
     Ok(())
 }
-
```
Comment (NGA-TRAN) [inline compactor2/src/driver.rs:134]: Move these tests to `compactor_tests.rs`

File: compactor2/src/compactor_tests.rs [hunk 2]
```diff
@@ -0,0 +1,182 @@
+#[cfg(test)]
```
Comment (NGA-TRAN) [inline compactor2/src/compactor_tests.rs:1]: 2 first tests were moved from `driver.rs`. The third one is newly added for this PR purpose

**Review by pauldix — APPROVED** [review]

File: compactor2/src/driver.rs [hunk 3]
```diff
@@ -57,6 +55,20 @@ async fn try_compact_partition(
         return Ok(());
     }
 
+    // Ignore partitions that are in skipped_compactions
```
Comment (crepererum) [inline compactor2/src/driver.rs:58]: This should be a filter. Adding an extra return will mess up all the metrics. I'll fix it

Comment (crepererum) [reply compactor2/src/driver.rs:58]: Done. See https://github.com/influxdata/influxdb_iox/pull/6673

**Pull request merged**